src/angular/components/lock.component.ts

```diff
@@ -46,12 +50,20 @@ export class LockComponent implements OnInit {
     async ngOnInit() {
         this.pinSet = await this.vaultTimeoutService.isPinLockSet();
         this.pinLock = (this.pinSet[0] && this.vaultTimeoutService.pinProtectedKey != null) || this.pinSet[1];
+        this.supportsBiometric = await this.platformUtilsService.supportsBiometric();
+        this.biometricLock = await this.vaultTimeoutService.isBiometricLockSet();
+        this.biometricText = await this.storageService.get(ConstantsService.biometricText);
         this.email = await this.userService.getEmail();
         let vaultUrl = this.environmentService.getWebVaultUrl();
         if (vaultUrl == null) {
             vaultUrl = 'https://bitwarden.com';
         }
         this.webVaultHostname = Utils.getHostname(vaultUrl);
+        this.router.routerState.root.queryParams.pipe(first()).subscribe(params => {
+            if (this.supportsBiometric && params['promptBiometric']) {
+                this.unlockBiometric();
+            }
+        });
```

Comment:
I don't think this is how we check query params in our other components. Search `this.route.queryParams.subscribe` in code. Not sure I follow why this scenario is different.

Reply:
We use it in vault.component, however that scenario is a bit different since the route actually changes and interacts with the component. In this case I was only interested in reading the queryParms once to decide if it should prompt a biometric unlock hence the use if `first()` which automatically closes the subscription after the first message.